R/tm_data_table.R

```diff
@@ -198,9 +206,8 @@ srv_page_data_table <- function(id,
       is.data.frame(isolate(data())[[name]])
     }, datanames)
 
-    if (!identical(datasets_selected, character(0))) {
-      checkmate::assert_subset(datasets_selected, datanames)
-      datanames <- datasets_selected
+    if (!identical(datanames, character(0))) {
+      checkmate::assert_subset(datanames, datanames)
```

Comment:
Argument `datanames` in `srv_`  is not needed anymore. If `tm_page_data_table(datanames)` is set to `a, b, c` then server would receive `a, b , c` packaged in `data()` object. Checking  if `datanames` is a subset of `names(data())` doesn't make sense because it probably will be (this is how teal works).  

```suggestion
```